contracts/root/depositManager/DepositManager.sol

```diff
@@ -79,47 +96,52 @@ contract DepositManager is DepositManagerStorage, IDepositManager, IERC721Receiv
     address wethToken = registry.getWethTokenAddress();
     WETH t = WETH(wethToken);
     t.deposit.value(msg.value)();
-    _createDepositBlock(msg.sender, wethToken, msg.value);
+    _safeCreateDepositBlock(msg.sender, wethToken, msg.value);
   }
 
   function depositERC721ForUser(address _token, address _user, uint256 _tokenId)
     public
   {
     ERC721(_token).transferFrom(msg.sender, address(this), _tokenId);
-    _createDepositBlock(_user, _token, _tokenId);
+    _safeCreateDepositBlock(_user, _token, _tokenId);
   }
 
   /**
-   * @notice This will be invoked when someone calls safeTransferFrom and deposits tokens to this contract
-     without directly interacting with this contract
-   * Note: the contract address is always the message sender.
-   * _operator The address which called `safeTransferFrom` function
-   * @param _from The address which previously owned the token
+   * @notice This will be invoked when safeTransferFrom is called on the token contract to deposit tokens to this contract
+     without directly interacting with it
+   * @dev msg.sender is the token contract
+   * _operator The address which called `safeTransferFrom` function on the token contract
+   * @param _user The address which previously owned the token
    * @param _tokenId The NFT identifier which is being transferred
    * _data Additional data with no specified format
    * @return `bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"))`
-   * unless throwing
    */
-  function onERC721Received(address /* _operator */, address _from, uint256 _tokenId, bytes memory /* _data */)
+  function onERC721Received(address /* _operator */, address _user, uint256 _tokenId, bytes memory /* _data */)
     public
     returns (bytes4)
   {
     // the ERC721 contract address is the message sender
-    _createDepositBlock(_from, msg.sender /* token */, _tokenId);
+    _safeCreateDepositBlock(_user, msg.sender /* token */, _tokenId);
     return 0x150b7a02;
   }
 
   // See https://github.com/ethereum/EIPs/issues/223
   function tokenFallback(address _user, uint256 _amount, bytes memory /* _data */)
   public
   {
-    _createDepositBlock(_user, msg.sender /* token */, _amount);
+    _safeCreateDepositBlock(_user, msg.sender /* token */, _amount);
   }
 
-  function _createDepositBlock(address _user, address _token, uint256 amountOrNFTId)
+  function _safeCreateDepositBlock(address _user, address _token, uint256 _amountOrToken)
     internal
     isTokenMapped(_token)
   {
-    rootChain.createDepositBlock(_user, _token, amountOrNFTId);
+    rootChain.createDepositBlock(_token, _amountOrToken, _user);
+  }
+
+  function _createDepositBlockBulk(address[] memory _tokens, uint256[] memory _amountOrTokens, address _user)
```

Comment:
where are we using it `_createDepositBlockBulk` ?

Reply:
to call `rootChain.bulkCreateDepositBlocks`

Reply:
Whoops missed it. Check it out now. I removed this function.